### PR TITLE
Vctr 61 implement real imagine conj

### LINKED ISSUE #32: Implement real, imagine, conj

### PR DESCRIPTION
**Pull request by JanosGit**
Added a few little fixes for issues that came up recently.

### CODE REVIEW

File: include/vctr/Expressions/Complex/Imag.h [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+};
+
+} // namespace vctr::Expressions
+
+namespace vctr
+{
+/** Extracts the imaginary parts of the complex source values.
+
+    In case you want to chain more computationally expensive expressions on the
+    complex source values before this expression, considering buffering the complex
```
Comment (TheWaveWarden) [inline include/vctr/Expressions/Complex/Imag.h]: consider* ;-)

File: include/vctr/Miscellaneous/CompilerSpecificAttributes.h [hunk 2]
```diff
@@ -37,3 +39,20 @@
 #define VCTR_FORCEDINLINE inline __attribute__ ((always_inline))
 #endif
 #endif
+
+
+#if (VCTR_GCC || VCTR_CLANG)
+
+#define VCTR_JOIN_DIAGNOSTICS_STRINGS(x,y) VCTR_TO_STRING (x ## y)
```
Comment (TheWaveWarden) [inline include/vctr/Miscellaneous/CompilerSpecificAttributes.h:46]: whitespace missing in function

**Review by TheWaveWarden — APPROVED** [review]
very nice! only nitpicky remarks from my side 😃

File: include/vctr/Miscellaneous/CompilerSpecificAttributes.h (hunk 2 above)
Comment (JanosGit) [reply include/vctr/Miscellaneous/CompilerSpecificAttributes.h:46]: declarations of function like macros don't work when there is a whitespace ;)

File: include/vctr/Expressions/Complex/Imag.h (hunk 1 above)
Comment (JanosGit) [reply include/vctr/Expressions/Complex/Imag.h]: Thanks, fixed ;)

File: include/vctr/Miscellaneous/CompilerSpecificAttributes.h (hunk 2 above)
Comment (TheWaveWarden) [reply include/vctr/Miscellaneous/CompilerSpecificAttributes.h:46]: Ah I didn't know!

**Pull request merged**